Core/UI/steering/steering.c

```diff
@@ -614,3 +614,177 @@ void handle_ptt_message(void) {
+                               14);
+  set_label_color_hv_feedbacks(primary_hv_feedback_status_last_state->feedback_sd_end,
+                               15);
+  set_label_color_hv_feedbacks(primary_hv_feedback_status_last_state->feedback_sd_out,
+                               16);
+  set_label_color_hv_feedbacks(primary_hv_feedback_status_last_state->feedback_sd_in,
+                               17);
+  set_label_color_hv_feedbacks(primary_hv_feedback_status_last_state->feedback_sd_bms,
+                               18);
+  set_label_color_hv_feedbacks(primary_hv_feedback_status_last_state->feedback_sd_imd,
+                               19);
+}
+
+void set_label_color_hv_errors(int label, int i) {
+  if (label) {
+    lv_obj_set_style_border_color(
+            hv_errors[i], lv_color_hex(COLOR_RED_STATUS_HEX), LV_PART_MAIN);
+    lv_obj_set_style_bg_color(hv_errors[i], lv_color_hex(COLOR_RED_STATUS_HEX),
+                              LV_PART_MAIN);
+  } else {
+    lv_obj_set_style_border_color(
+            hv_errors[i], lv_color_hex(COLOR_GREEN_STATUS_HEX), LV_PART_MAIN);
+    lv_obj_set_style_bg_color(
+            hv_errors[i], lv_color_hex(COLOR_GREEN_STATUS_HEX), LV_PART_MAIN);
+  }
+}
+
+void hv_errors_update() {
+  // could be used enum_to_string but same outcome with still same number of
+  // lines
+  GET_LAST_STATE(primary, hv_errors, PRIMARY, HV_ERRORS);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_cell_low_voltage, 0);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_cell_under_voltage, 1);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_cell_over_voltage, 2);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_cell_high_temperature,
+                            3);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_cell_over_temperature,
+                            4);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_over_current, 5);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_can, 6);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_int_voltage_mismatch,
+                            7);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_cellboard_comm, 8);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_cellboard_internal, 9);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_connector_disconnected,
+                            10);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_fans_disconnected, 11);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_feedback, 12);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_feedback_circuitry, 13);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_eeprom_comm, 14);
+  set_label_color_hv_errors(primary_hv_errors_last_state->errors_eeprom_write, 15);
+}
+
+void set_label_color_das_errors(bool label, int i) {
```

Comment:
`set_label_color_das_errors`, `set_label_color_hv_errors`, `set_label_color_lv_errors` are all the same. Also `set_label_color_hv_feedbacks` is similar. Try to write only one function to to all this. The simpler is the function, the better.